### PR TITLE
Add SDPA_FW TTML op

### PR DESCRIPTION
**Pull request by djstefanovicTT**
### Ticket
Part of https://github.com/tenstorrent/tt-mlir/issues/9097

### Problem description
Unimplemendted TTML op in compiler.

### What's changed
Added composite ops and relevant conversions for SDPA_FW op.

### Checklist
- [x] New/Existing tests provide coverage for changes

### CODE REVIEW

File: lib/Dialect/TTNN/IR/TTNNOps.cpp [hunk 1]
```diff
@@ -3701,6 +3701,37 @@ ::mlir::LogicalResult mlir::tt::ttnn::AdamWOp::verify() {
...
+  if (maskType != ttcore::AttentionMaskType::Arbitrary && getAttentionMask()) {
+    return emitOpError(
+        "attention_mask is only allowed when mask_type is 'arbitrary'");
+  }
+
+  if (getReturnIntermediates() != static_cast<bool>(getIntermediates())) {
+    return emitOpError("intermediates result must be present iff "
+                       "return_intermediates is true");
+  }
+
```
[the remark this answers is not in the document]
Comment (djstefanovicTT) [reply lib/Dialect/TTNN/IR/TTNNOps.cpp:3811]: Validation is done in decomposition.

File: lib/Conversion/StableHLOToTTIR/StableHLOLegalizeCompositePass.cpp [hunk 2]
```diff
@@ -2009,6 +2009,78 @@ class TenstorrentAdamWConversionPattern
...
+
+    DictionaryAttr compositeAttrs = srcOp.getCompositeAttributes();
+
+    // mask_type is an integer selector (0=None, 1=Causal, 2=Arbitrary),
+    // defaulting to Causal.
+    ttcore::AttentionMaskType maskType = ttcore::AttentionMaskType::Causal;
+    if (compositeAttrs) {
+      if (auto maskAttr = mlir::dyn_cast_or_null<IntegerAttr>(
+              compositeAttrs.get("mask_type"))) {
+        maskType = static_cast<ttcore::AttentionMaskType>(maskAttr.getInt());
```
Comment (mtopalovicTT) [inline lib/Conversion/StableHLOToTTIR/StableHLOLegalizeCompositePass.cpp]: Assert that mask attribute is within bounds of the enum [0, 2].

Comment (djstefanovicTT) [reply lib/Conversion/StableHLOToTTIR/StableHLOLegalizeCompositePass.cpp]: Updated.

**Review by pglusacTT — APPROVED** [review]
LGTM

**Review by pilkicTT — APPROVED** [review]

**Review by mtopalovicTT — APPROVED** [review]

**Review by jgrimTT — APPROVED** [review]

**Review by mmanzoorTT — APPROVED** [review]
